### PR TITLE
verbose mode and select sections for annotate command

### PR DESCRIPTION
**Pull request by gotcha**
--verbose allows, among others, to find out which values are overridden
when extending cfg files

--section allows to focus on a specific section

### CODE REVIEW

File: doc/reference.rst [hunk 1]
```diff
@@ -117,6 +117,13 @@ where they came from. Try it!
 
 .. _bootstrap-command:
 
+
+``--verbose``
+  Increase the verbosity of the output: display all steps that compute the final values used by buildout.
+
+``--section section``
+  Only displya annotation for the given section.
```
Comment (hvelarde) [inline doc/reference.rst]: "display"

File: doc/reference.rst [hunk 2]
```diff
@@ -117,6 +117,13 @@ where they came from. Try it!
 
 .. _bootstrap-command:
 
+
+``--verbose``
```
Comment (jimfulton) [inline doc/reference.rst]: Why not control this via Buildout's verbosity setting?

File: doc/reference.rst [hunk 3]
```diff
@@ -117,6 +117,13 @@ where they came from. Try it!
...
+  Increase the verbosity of the output: display all steps that compute the final values used by buildout.
+
+``--section section``
```
Comment (jimfulton) [inline doc/reference.rst]: Why --section?

Why not just list one or more sections?

File: src/zc/buildout/buildout.py [hunk 4]
```diff
@@ -1106,7 +1234,16 @@ def setup(self, args):
...
-        _print_annotate(self._annotated)
+        if args and "--verbose" in args:
+            verbose = True
+        else:
+            verbose = False
+        section = None
+        if args:
+            for arg in args:
+                if arg.startswith('--section'):
+                    _, section = arg.split("=")
```
Comment (jimfulton) [inline src/zc/buildout/buildout.py]: I don't see mention of '=' in reference.

if the option is repeated, the last one wins? :(

I don't think flags are needed. Buildout's verbosity (aka log level) should be used.

Why not just accept a list of sections?  If TCBOO, then error if more than one is provided.

**Comment by gotcha** [comment]
@jimfulton Thanks for your review and suggestions. I implemented both.

**Comment by jimfulton** [comment]
I haven't done a thorough review. Those were just drive-by comments. :)

**Comment by gotcha** [comment]
See https://github.com/buildout/buildout/pull/378/commits/5e018a2fa086fc899baeae12391bfde0562590c3#diff-166a7a24d7700d690dd9ee5ee82ad04b

**Comment by gotcha** [comment]
Anyone else for a review ? Or should I merge ?

**Comment by gotcha** [comment]
All tests pass. And I let enough time pass for reviews.

**Pull request merged**